ddl/ddl_api.go

```diff
@@ -3363,6 +3381,17 @@ func (d *ddl) AlterTable(ctx context.Context, sctx sessionctx.Context, stmt *ast
 						Name: model.NewCIStr(opt.StrValue),
 					}
 				case ast.TableOptionEngine:
+				case ast.TableOptionTTL:
+					var ttlInfo *model.TTLInfo
+					ttlInfo, err = buildTTLInfo(opt)
+					if err != nil {
+						return err
+					}
+					err = d.AlterTableTTLInfo(sctx, ident, ttlInfo)
+				case ast.TableOptionTTLEnable:
+					err = d.AlterTableTTLEnable(sctx, ident, opt.BoolValue)
```

Comment:
I think it's better to move these two operations to one DDL job for atomic reason.

Reply:
Yes. After combining `TTLInfo` and `TTLEnable` together, they have to be done together.

And I also make sure a single `alter` statement will only cause one alter ttl ddl job (by only respecting the last configuration on `TTL` and `TTL_ENABLE` ).